aiohttp/connector.py

```diff
@@ -125,7 +125,13 @@ def __init__(self, *, conn_timeout=None, keepalive_timeout=30,
         self._conns = {}
         self._acquired = defaultdict(set)
         self._conn_timeout = conn_timeout
-        self._keepalive_timeout = keepalive_timeout
+        if force_close and keepalive_timeout:
```

Comment:
```
marker = object()

class BaseConnector:

    def __init__(self, keepalive_timeout=marker, force_close=marker):

        if keepalive_timeout is not marker and force_close is not marker:
            # if keepalive and force_close are something different rather than object class instance
            if keepalive_timeout and force_close:
                # if both keepalive and force_close were passed as positive values
                raise ValueError("Can't accept both 'force_close=True' "
                                 "and 'keepalive_timeout={0}'"
                                 .format(keepalive_timeout))
            else:
                if keepalive_timeout == 0.0:
                    # it may appear that keepalieve was passed as 0.0 and force_close as False
                    self.force_close = True
                self.keepalive_timeout = keepalive_timeout
        elif keepalive_timeout is not marker:
            # if only keepalive was passed as something different rather than object class instance
            self.force_close = False
            if not isinstance(keepalive_timeout, float):
                # if keepalive is not a float
                if isinstance(keepalive_timeout, type(None)):
                    # possibly keepalive can be NoneType
                    self.keepalive_timeout = 0.0
                else:
                    # if keepalive is not float and not NoneType
                    raise TypeError("keepalive_timeout: float required.")
            else:
                self.keepalive_timeout = keepalive_timeout
        elif force_close is not marker:
            # if force_close is something different from object class instance
            if not isinstance(force_close, bool):
                raise TypeError("force_close: bool required.")
            else:
                self.force_close = force_close
                self.keepalive_timeout = 0.0 if force_close else 30.0
        elif keepalive_timeout is marker and force_close is marker:
            # if keepalive was not passed - using default value (30.0 secs)
            # and force_close set to False
            self.keepalive_timeout = 30.0
            self.force_close = False


if __name__ == "__main__":

    print(BaseConnector().keepalive_timeout)
    print(BaseConnector(force_close=False).keepalive_timeout)
    print(BaseConnector(force_close=True).keepalive_timeout)
    print(BaseConnector(keepalive_timeout=1.0).force_close)
    print(BaseConnector(keepalive_timeout=0.0, force_close=False).keepalive_timeout)
    print(BaseConnector(keepalive_timeout=5.0, force_close=False).keepalive_timeout)
    try:
        print(BaseConnector(keepalive_timeout=1).force_close)
        print(BaseConnector(force_close='bbb').force_close)
    except Exception as e:
        print(str(e))

```

After investigations it seems like given code does what you were asking for, @asvetlov .
As i can see this code snippet doesn't break backward compatibility (see main section in code above).

Please let me know if that would work for us.